pipelines/controller/src/main/java/org/openmrs/analytics/PipelineManager.java

```diff
@@ -304,6 +307,85 @@ synchronized void runIncrementalPipeline()
     }
   }
 
+  /**
+   * This method checks upon Controller start checks on Thrift sever to create resource tables if
+   * they don't exist. There is a @PostConstruct method present in this class which is initDwhStatus
+   * and the reason below code has not been added because dataProperties.getThriftserverHiveConfig()
+   * turns out to be null when used by
+   * DatabaseConfiguration.createConfigFromFile(dataProperties.getThriftserverHiveConfig()).
+   */
+  @Override
+  public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
```

Comment:
Can you please add a button to the UI (and an API endpoint) for triggering this logic? So the idea is that beside doing this on start-up (which is great) we also give the user the option of triggering it manually (or with an API call) if they restart the Metastore for whatever reason.

Reply:
done